src/stream/src/executor/top_n/top_n_cache.rs

```diff
@@ -181,6 +187,21 @@ impl<const WITH_TIES: bool> TopNCache<WITH_TIES> {
         self.high.clear();
     }
 
+    /// Get total count of entries in the cache.
+    pub fn len(&self) -> usize {
+        self.low.len() + self.middle.len() + self.high.len()
+    }
+
+    pub(super) fn update_table_row_count(&mut self, table_row_count: usize) {
+        self.table_row_count = Some(table_row_count)
+    }
+
+    fn table_row_count_matched(&self) -> bool {
```

Comment:
May add some comments about what it implies when they match or mismatch

Reply:
In `fill_high_cache` we scan the whole group in the table, skipping rows that below `high`, this PR counts these skipped row as well.

However e2e failed, I'm still debugging.